static/js/pages/storages/StorageForm.tsx

```diff
@@ -0,0 +1,160 @@
+    name: Yup.string()
+      .test(
+        "deduplicate",
+        "A storage pool with this name already exists",
+        (value) =>
+          checkDuplicateName(
+            value,
+            panelParams.project,
+            controllerState,
+            "storage-pools"
+          )
+      )
+      .required("This field is required"),
+  });
+
+  const formik = useFormik({
+    initialValues: {
+      name: "",
+      description: "",
+      driver: "zfs",
+      source: "",
+      size: "",
+    },
+    validationSchema: SnapshotSchema,
+    onSubmit: ({ name, description, driver, source, size }) => {
+      const storagePool: LxdStorage = {
+        name,
+        description,
+        driver,
+        source: driver !== "btrfs" ? source : undefined,
+        config: {
+          size: size ? `${size}GiB` : undefined,
+        },
+      };
+
+      createStoragePool(storagePool, panelParams.project)
+        .then(() => {
+          void queryClient.invalidateQueries({
+            queryKey: [queryKeys.storage],
+          });
+          panelParams.clear();
+        })
+        .catch((e) => {
+          formik.setSubmitting(false);
+          notify.failure("Error on storage pool creation.", e);
+        });
+    },
+  });
+
+  return (
+    <Aside>
+      <div className="p-panel">
+        <PanelHeader title={<h4>Create storage pool</h4>} />
+        <div className="p-panel__content">
+          <NotificationRow notify={notify} />
+          <Row>
+            <Form onSubmit={formik.handleSubmit} stacked>
+              <Input
+                id="name"
+                name="name"
+                type="text"
+                label="Name"
+                onBlur={formik.handleBlur}
+                onChange={formik.handleChange}
+                value={formik.values.name}
+                error={formik.touched.name ? formik.errors.name : null}
+                required
+                stacked
+              />
+              <Input
+                id="description"
+                name="description"
+                type="text"
+                label="Description"
+                onBlur={formik.handleBlur}
+                onChange={formik.handleChange}
+                value={formik.values.description}
+                error={
+                  formik.touched.description ? formik.errors.description : null
+                }
+                stacked
+              />
+              <Select
+                name="driver"
+                help="ZFS gives best performance and reliability"
+                label="Driver"
+                options={storageDrivers}
+                onChange={formik.handleChange}
+                value={formik.values.driver}
+                required
+                stacked
+              />
+              <Input
+                id="size"
+                name="size"
+                type="number"
+                help="Optional, defaults to 20% of free disk space, >= 5 GiB and <= 30 GiB"
```

Comment:
I'm confused, the value should be between 5 and 30? Are decimals allowed? (I already know the answer from what the BE returns: no)
The point is that I tried with values lower than 5 and it succeeds. With 31 also succeeds. Where do those extremes come from?
Maybe some constraints should be added by using the `min`, `max` and `step` properties.

Reply:
This is from the docs, the default is 20%, but in that range. I'll remove all the help text apart from optional.

Reply:
That "Optional" alone seems out of place; the "Description" is also optional, but we don't have a help text below it just to say that it is optional.

Reply:
Right, I circled it back to "When left blank, defaults to 20% of free disk space. Default will be between 5GiB and 30GiB". Hope it is less confusing?

Reply:
A bit less, but still confusing. I cannot think of a better wording though, so let's keep it as it is for the time being, and maybe iterate on it later.